### PR TITLE
chore(glue): timeout and worker type validation for Ray jobs

### LINKED ISSUE #29612: (glue): Timeout and workerType missing validation for Ray job types
### Describe the bug

When developing Ray jobs via CDK, I bumped into the fact that specifying a timeout or not specifying a worker type on `Job` for Ray job types will succeed in synthesis, but fail deployment.

### Expected Behavior

I'd expect an error to be thrown if workerType is not provided or timeout is provided, as both of these will cause deployment failures for Ray job types.

### Current Behavior

Synthesis passes and failure occurs on deployment: 

```
 UPDATE_FAILED [...] Timeout not supported for Ray jobs
```
or 
```
CREATE_FAILED [...] Worker type cannot be null and only [Z.2X] worker types are supported for glueray jobs
```

### Reproduction Steps

```
#!/usr/bin/env node
import * as cdk from 'aws-cdk-lib';
import { Code, GlueVersion, Job, JobExecutable, PythonVersion, Runtime, WorkerType } from '@aws-cdk/aws-glue-alpha';

const app = new cdk.App();
const stack = new cdk.Stack(app, 'glue-job-issue-stack');
const job = new Job(stack, 'MyJob', {
  jobName: 'glue-issue-test-job',
  executable: JobExecutable.pythonRay({
    glueVersion: GlueVersion.V4_0,
    pythonVersion: PythonVersion.THREE_NINE,
    runtime: Runtime.RAY_TWO_FOUR,
    script: Code.fromAsset('lib/test-job.py')
  }),
  workerType: WorkerType.G_025X,
  workerCount: 1,
  timeout: cdk.Duration.minutes(10),
});
```

### Possible Solution

Additional validation in `packages/@aws-cdk/aws-glue-alpha/lib/job.ts` would address this and provide a slightly smoother developer experience when working with ray job types. I would be interested in contributing a fix if triage determines this is valid bug.

### Additional Information/Context

_No response_

### CDK CLI Version

2.133.0

### Framework Version

2.133.0

### Node.js Version

20.10.0

### OS

macOS

### Language

TypeScript

### Language Version

5.3.3

### Other information

_No response_

### PR DESCRIPTION
**Pull request by badmintoncryer**
### Issue # (if applicable)

Closes #29612.

### Reason for this change

AWS Glue Ray job has some restriction.
- must use Z.2X worker type
```sh
CREATE_FAILED [...] Worker type cannot be null and only [Z.2X] worker types are supported for glueray jobs
```
- must not specify timeout
```sh
UPDATE_FAILED [...] Timeout not supported for Ray jobs
```

### Description of changes

Add validation for above restriction.

```ts
    if (executable.type.name === JobType.RAY.name) {
      if (props.workerType !== WorkerType.Z_2X) {
        throw new Error(`WorkerType must be Z_2X for Ray jobs, got: ${props.workerType}`);
      }
      if (props.timeout !== undefined) {
        throw new Error('Timeout cannot be set for Ray jobs');
      }
    }
```

### Description of how you validated changes

Add unit test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by mazyu36 — APPROVED** [review]
The changes are LGTM. 
However, this fix will not be reflected once #30833 is merged. 
I'll leave the decision on how to proceed to the maintainers.

**Comment by badmintoncryer** [comment]
@mazyu36 Thanks for the information. I don't know this PR. I'll wait for the maintainer's opinion.

**Review by GavinZZ — APPROVED** [review]

**Pull request merged**
